src/nexus.cpp

```diff
@@ -160,31 +152,35 @@ void Nexus::start()
  */
 void Nexus::showLogin()
 {
+    loginScreen = new LoginScreen();
+
+#ifdef Q_OS_MAC
+    connect(loginScreen, &LoginScreen::windowStateChanged, this,
+            &Nexus::onWindowStateChanged);
+#endif
+
     delete widget;
     widget = nullptr;
 
     delete mProfile;
     mProfile = nullptr;
 
     loginScreen->reset();
-    loginScreen->move(QApplication::desktop()->screen()->rect().center() - loginScreen->rect().center());
     loginScreen->show();
-    ((QApplication*)qApp)->setQuitOnLastWindowClosed(true);
+    loginScreen->move(QApplication::desktop()->screen()->rect().center() -
```

Comment:
@antis81 After reading the linked documentation, I think the correct approach would be to call `show()` after `move()` because it will apparently not fail as I thought. I think @Diadlo is right that it'll provide a more glitch free experience.